jax_distribution/systems/ff_ippo_rware.py

```diff
@@ -317,15 +319,133 @@ def learner_fn(runner_state: RunnerState) -> ExperimentOutput:
             _update_step, in_axes=(0, None), axis_name="batch"
         )
 
-        runner_state, metric = jax.lax.scan(
-            batched_update_step, runner_state, None, config["NUM_UPDATES"]
+        runner_state, (metric, loss_info) = jax.lax.scan(
+            batched_update_step, runner_state, None, config["NUM_UPDATES_PER_EVAL"]
         )
-        return {"runner_state": runner_state, "metrics": metric}
+        total_loss, (value_loss, loss_actor, entropy) = loss_info
+        return {
+            "runner_state": runner_state,
+            "metrics": metric,
+            "total_loss": total_loss,
+            "value_loss": value_loss,
+            "loss_actor": loss_actor,
+            "entropy": entropy,
+        }
 
     return learner_fn
 
 
-def learner_setup(env: Environment, config: Dict) -> Tuple[callable, RunnerState]:
+def get_evaluator_fn(env: Environment, apply_fn: callable, config: dict) -> callable:
+    """Get the evaluator function."""
+
+    def eval_one_episode(params, runner_state) -> Tuple:
+        """Evaluate one episode. It is vectorized over the number of evaluation episodes."""
+
+        def _env_step(runner_state: Tuple) -> Tuple:
+            """Step the environment."""
+            # PRNG keys.
+            rng, env_state, last_timestep, step_count_, return_ = runner_state
+
+            # Select action.
+            rng, _rng = jax.random.split(rng)
+            pi, _ = apply_fn(params, last_timestep.observation)
+
+            if config["EVALUATION_GREEDY"]:
+                action = pi.mode()
+            else:
+                action = pi.sample(seed=_rng)
+
+            # Step environment.
+            env_state, timestep = env.step(env_state, action)
+
+            # Log episode metrics.
+            return_ += timestep.reward
+            step_count_ += 1
+            runner_state = (rng, env_state, timestep, step_count_, return_)
+            return runner_state
+
+        def is_done(carry: Tuple) -> jnp.bool_:
```

Comment:
As mentioned previous avoid the `jnp.xyz` for types. Use chex.Numeric/jax.typing.ArrayLike or in this case just `bool`

```suggestion
        def is_done(carry: Tuple) -> bool:
```